File: 3397-maximum-number-of-distinct-elements-after-operations/3397-maximum-number-of-distinct-elements-after-operations.go
func maxDistinctElements(nums []int, k int) int {
    sort.Ints(nums)
    var curr int = nums[0] - k
    var n int = len(nums)
    var result int = 1
    for i := 1; i < n; i ++ {
        if (curr < nums[i] + k) {
            result ++
            curr = max(nums[i] - k, curr + 1)
        }
    }
    return result
}

File: 3397-maximum-number-of-distinct-elements-after-operations/3397-maximum-number-of-distinct-elements-after-operations.py
class Solution:
    def maxDistinctElements(self, nums: List[int], k: int) -> int:
        nums.sort()
        curr = nums[0] - k
        result = 1
        n = len(nums)
        for i in range(1, n):
            if nums[i] + k > curr:
                result += 1
                curr = max(curr + 1, nums[i] - k)
        return result